src/main/java/com/aws/greengrass/mqtt/bridge/BridgeConfig.java

```diff
@@ -197,6 +199,33 @@ private static long getSessionExpiryInterval(Topics configurationTopics) {
         return sessionExpiryInterval;
     }
 
+    /**
+     * Get mqtt5 route options for a {@link com.aws.greengrass.mqtt.bridge.TopicMapping.TopicType},
+     * grouped by topic names.
+     *
+     * <p>If bridge is not configured for mqtt5, no options will be returned.
+     *
+     * @param source source type
+     * @return route options by topic name
+     */
+    public Map<String, Mqtt5RouteOptions> getMqtt5RouteOptionsForSource(TopicMapping.TopicType source) {
+        if (mqttVersion != MqttVersion.MQTT5) {
+            return Collections.emptyMap();
+        }
+        Map<String, Mqtt5RouteOptions> opts = new HashMap<>();
+        for (String route : topicMapping.keySet()) {
+            TopicMapping.MappingEntry mapping = topicMapping.get(route);
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

**Problem**: Iterating over [map.keySet()](https://docs.oracle.com/javase/8/docs/api/java/util/Map.html#keySet--) and calling [map.get(Key)](https://docs.oracle.com/javase/8/docs/api/java/util/Map.html#get-java.lang.Object-) to get the values within the loop is inefficient. 
**Fix**: Iterate over [map.entrySet()](https://docs.oracle.com/javase/8/docs/api/java/util/Map.html#entrySet--) and access the values using [entry.getValue()](https://docs.oracle.com/javase/8/docs/api/java/util/Map.Entry.html#getValue--) to avoid potential performance penalty due to lookup.